### DISCUSSION TITLE
Unable to flash Gosund P1 power strip via serial

### DISCUSSION THREAD

**Discussion by eu-gh** [arendst/Tasmota#22018] [Q&A]
Hi!

I recently bought said Tuya based product, hoping to make it actually usable in my home environment.

The built-in communication board is a "CUCO_Z0_V1.1", which, according to the Internet™, is still ESP-based, so I tried tuya-convert, which failed due to the base firmware being too recent.

So I tried connecting via serial, which is easy enough on this device (https://forum.iobroker.net/topic/36086/gosund-p1-1-steckdose-mit-tasmota-flashen-ota/15), but I can't seem to connect to the chip at all. When checking the serial console (using default 115200 8N1 settings), I only get a little bit of garbage and, of course, esptool doesn't work under these circumstances. Connecting IO0 to GND also doesn't change anything, apart from shutting off the blue LED.
![IMG_20240825_155844](https://github.com/user-attachments/assets/3f3a3d31-eccb-472c-b3bf-6064a006b2ae)

I used a RasPI onboard serial, which worked with other devices, but failed here, then found the information that it is "not possible" to use the +3.3V supply voltage directly (why?), so I shoehorned a voltage regulator in between the 5V output, which made no difference. I then tried using a regular ESP development board with RST pin connected to GND, but had the exact same result.

So is this a voltage issue? Or is there anything else I can try? The soldering connections seems to be solid.

Thanks and best regards!

**Comment by s-hadinger** [comment]
Can you make a picture of the cpu?

**Reply by barbudor** [reply]
> I then tried using a regular ESP development board with RST pin connected to GND, but had the exact same result.

This seems to confirm something is wrong in your process such as connecting RST to GND
GPIO0 should be connected to GND
RST should only be pulsed to GND and releases otherwise the ESP doesn't start

**Reply by eu-gh** [reply]
> Can you make a picture of the cpu?

It's shielded unfortunately.
![PXL_20240823_233726313](https://github.com/user-attachments/assets/0b09fedd-2731-46d0-b1d8-9499917c817b)
![PXL_20240823_233739859](https://github.com/user-attachments/assets/f6a669f5-8a64-42d5-8436-c794eb2f1993)



> This seems to confirm something is wrong in your process such as connecting RST to GND GPIO0 should be connected to GND RST should only be pulsed to GND and releases otherwise the ESP doesn't start

I'm using the second ESP just as a USB <-> serial bridge, it's not supposed to start. This setup is working fine for connecting to other ESPs, but from the power strip, I'm only getting random characters like

`d��l��dp���n����cd$��c`

**Reply by sfromis** [reply]
**If** those "random characters" are data sent by the CSE7766 energy monitoring chip (which is connected to ESP8266 standard serial pins gpio 1/3), you seeing activity on your receive pin might imply that you have swapped TX/RX, as you should be connecting your serial RX to the TX pin of the ESP8266, where the energy monitoring chip should not be transmitting.

This is assuming that you have correctly booted the ESP8266 in flashing mode, by having gpio0 grounded before/while applying power.

Still, **if** that CSE7766 is actively sending data "all the time", that could be bad for flashing.

**Reply by eu-gh** [reply]
You are right - they were initially connected the right way, but during testing I swapped the wires at some point. Still, this is the only way I could get some sign of life out of this thing, with the correct wiring, the console stays completely blank and esptool always tells me that no serial data was received.

**Reply by rakaiser** [reply]
I've an GHome P1 from Amazon.de which is the same power strip. It comes with the same CUCO Z V1.1 device (with the small V1.4 above) and **I can confirm that this is an ESP device** (ESP8285 with 1MB flash). 
Programming with an USB serial dongle and Tasmotizer was unproblematic.  These is no interference from the power measurement chip as it is not powered at all and backpath the isolation chip found at backside of PCB obviously idle.
ESP and the USB are  on isolated 5V whereas the CSE7766 may be on the hot AC side (thus the many cut-outs in the PCB as isolation barrier) .

**Comment by Jason2866** [comment]
I would surprised if the used module has an espressif MCU inside. **All** actual Tuya devices do not use espressif MCUs.

**Reply by sfromis** [reply]
The templates repository page about the ESP8266-based [Gosund P1](https://templates.blakadder.com/gosund_P1.html) does show the "CUCO-Z0-V1.1" module, like in the photos in this thread. And the Pinouts page also lists [CUCO-Z0-V1.1](https://tasmota.github.io/docs/Pinouts/#cuco-z0-v11-used-by-gosund).

**Reply by Jason2866** [reply]
Do you make bets the do not change inside without changing name?
Since flashing is not working everything should be checked.
